carmin.yaml

```diff
@@ -266,6 +265,133 @@ paths:
           description: successful response
         default:
           $ref: '#/responses/GenericError'
+  /path/{uri}:
```

Comment:
Realy using the uri on the URL ? I think there are too many characters to escape in a uri, especially the slashs, and it would be a big burden to implement. I've used a query parameter in my "alpha" implementation.

Reply:
You mean, the URL as in:
* URI: shablanc://shablanc.somedomain.org/home/foo/file.txt
* URL: /home/foo/file.txt
?

Reply:
I mean as in `https://carmin-server.com/rest/path/{uri}`. If the path uri is `scheme://server.com/path/to/file` then the rest URL to get it would be `https://carmin-server.com/rest/path/scheme://server.com/path/to/file` with all reserved characters in the file uri escaped. That would be very cumbersome.

That's why I choose to put the uri in a query parameter in the VIP implementation, like `https://carmin-server.com/rest/path?uri=scheme://server.com/path/to/file`, even if it isn't ideal

Reply:
I see. Then the URI should be `https://carmin-server.com/rest/path/<local_path>` and we should add `<local_path>` to the Path model so that we can use it in the query. `<local_path>` is neither a URI nor a URL.

Reply:
With <local_path> being slash separated like "path/to/file" ? After encoding it would be `https://carmin-server.com/rest/path/path%2Fto%2Ffile`. I know i struggled to accept this kind of things in VIP (slash in pipeline identifier) and settled a quite ugly solution that could not work for that. I could dig deeper of course, but the idea is that slashes in the path section of an URI may be difficult to handle in rest servers of various technologies (it is in spring).

Reply:
Then shall we leave the "local_id" completely up to the platform to define?

Reply:
Seems like a good idea indeed, easy to implement. But it makes the uri kind of useless. And it doesn't work for path existence & upload.
To discuss on Sept 6 I think.